### PR TITLE
Dockerfile based on Alpine

### PR DESCRIPTION
**Pull request by solipsist01**
Dockerfile to create a docker container.
Perhaps you could link your github to hub.docker.com and create an automated docker build from this dockerfile. Every time you update code, docker hub will rebuild the docker image etc. If you need help with that, i'd be glad to help.

### CODE REVIEW

File: Dockerfile [hunk 1]
```diff
@@ -0,0 +1,22 @@
+FROM mhart/alpine-node:base
```
Comment (andrewwippler) [inline Dockerfile]: It is best practice to use the official node base image. Who knows when/if this base image will get security patches.

I would suggest using `FROM node:8.11-alpine`

File: Dockerfile [hunk 2]
```diff
@@ -0,0 +1,22 @@
...
+
+ENV PATH /root/.yarn/bin:$PATH
```
Comment (andrewwippler) [inline Dockerfile]: This is not needed on the official base image

File: Dockerfile [hunk 3]
```diff
@@ -0,0 +1,22 @@
...
+
+RUN apk update \
+  && apk add curl bash binutils tar git \
+  && rm -rf /var/cache/apk/* \
```
Comment (andrewwippler) [inline Dockerfile]: `apk add --no-cache curl bash binutils tar git` does the same thing as the above 3 lines. 

Is there a reason why bash must be used? `sh` should suffice for a nodejs app.

File: Dockerfile [hunk 4]
```diff
@@ -0,0 +1,22 @@
...
+  && /bin/bash \
+  && touch ~/.bashrc \
```
Comment (andrewwippler) [inline Dockerfile]: What is in `~/.bashrc`?

File: Dockerfile [hunk 5]
```diff
@@ -0,0 +1,22 @@
...
+  && curl -o- -L https://yarnpkg.com/install.sh | bash \
```
Comment (andrewwippler) [inline Dockerfile]: `yarn` is in the node 8+ base images.

File: Dockerfile [hunk 6]
```diff
@@ -0,0 +1,22 @@
...
+  && git clone https://github.com/rage8885/OpenSC2K \
```
Comment (andrewwippler) [inline Dockerfile]: With the Dockerfile being in the repository, git cloning is not needed. We can just `ADD . /home/node/app`

File: Dockerfile [hunk 7]
```diff
@@ -0,0 +1,22 @@
...
+  && cd OpenSC2K \
+  && yarn --ignore-engines install \
```
Comment (andrewwippler) [inline Dockerfile]: Not sure if the flag is needed in the official base image.

File: Dockerfile [hunk 8]
```diff
@@ -0,0 +1,22 @@
...
+  && echo 'cd /OpenSC2K && yarn run start' >/run.sh \
+  && chmod +x /run.sh \
+  && apk del git curl tar binutils
+  
+EXPOSE 3000
+
+ENTRYPOINT [ "/bin/bash" ]
+
+CMD [ "/run.sh" ]
```
Comment (andrewwippler) [inline Dockerfile]: If adding the files to the `/home/node/app` in the node:8+ base images, the working directory should be `/home/node/app`. Essentially all that would need to be run (after the container is built) is `yarn run start`. That would look like `CMD ["yarn", "run", "start"]`.

File: Dockerfile [hunk 9]
```diff
@@ -0,0 +1,22 @@
...
+  && git clone https://github.com/rage8885/OpenSC2K \
+  && cd OpenSC2K \
+  && yarn --ignore-engines install \
+  && echo 'cd /OpenSC2K && yarn run start' >/run.sh \
+  && chmod +x /run.sh \
+  && apk del git curl tar binutils
+  
+EXPOSE 3000
+
+ENTRYPOINT [ "/bin/bash" ]
```
Comment (andrewwippler) [inline Dockerfile]: Entrypoints should be primarily used for running a script that parses ENV variables/etc. to create a dynamic config file or for pre-processing a file inside the container before it is run. `/bin/bash` is not one of those scripts.

**Comment by rage8885** [comment]
Thank you for the PR, but will rework this a bit and include in a future release. Right now without the assets included, you can't do anything with the game when launched.

**Pull request closed without merging**